docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,224 @@
+  </PropertyGroup>
+</Project>
+```
+
+### 3. Update Your Code
+
+- **Replace Deprecated Methods**: Update deprecated methods to newer versions.
+  
+  - **Assert.AreEqual/AreNotEqual (with object)** → Use generic versions.
+
+    **Before** (deprecated):
+
+    ```csharp
+    Assert.AreEqual(expectedObject, actualObject);
+    Assert.AreNotEqual(expectedObject, actualObject);
+    ```
+
+    **After** (using generics):
+
+    ```csharp
+    Assert.AreEqual<object>(expectedObject, actualObject);
+    Assert.AreNotEqual<object>(expectedObject, actualObject);
+    ```
+
+  - **Assert.AreSame/AreNotSame (with object)** → Use generic versions.
+
+    **Before** (deprecated):
+
+    ```csharp
+    Assert.AreSame(expectedObject, actualObject);
+    Assert.AreNotSame(expectedObject, actualObject);
+    ```
+
+    **After** (using generics):
+
+    ```csharp
+    Assert.AreSame<object>(expectedObject, actualObject);
+    Assert.AreNotSame<object>(expectedObject, actualObject);
+    ```
+
+- **Test Initialization**: Use `TestInitialize` methods for async initialization.
+- **Cleanup**: Use `TestCleanup` methods or the `Dispose` pattern for cleanup.
+- **RunSettings**: The `.testsettings` file is no longer supported, meaning `<LegacySettings>` is also no longer available. Use [.runsettings](https://learn.microsoft.com/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022) for test configuration.
+
+---
+
+## New Features in MSTest v3
+
+- Improved defaults for test projects
+- Simplified setup and usage
+- Enhanced extensibility of the MSTest runner
+- New Roslyn-based code analyzers for improved test development
+- Support for WinUI applications
+- In-assembly parallel execution
+- Dynamic data sources for data-driven tests
+
+---
+
+## Deprecated Features
+
+- Dropped support for:
+  - **.NET Framework 4.5** (use .NET 4.6.2 or higher)
+  - **.NET Standard 1.0** (use .NET Standard 2.0)
+  - **UWP versions before 16299**
+  - **WinUI versions before 18362**
+  - **.NET 5** (use .NET Core 3.1 or .NET 6)
+
+---
+
+## Breaking Changes and Removed APIs
+
+### Assertion Overloads
+
+MSTest v3 removes certain `Assert` overloads that accept object types to promote type-safe assertions. Update tests using these overloads to specify types explicitly.
+
+**Example**:
+
+```csharp
+// Old (v0/v1)
+Assert.AreEqual(expected, actual); // both are objects
+
+// New (v3)
+Assert.AreEqual<int>(expectedInt, actualInt); // specify the type
+```
+
+### DataRowAttribute Updates
+
+The `DataRowAttribute` constructors have been simplified in MSTest v3. Update parameterized tests using `DataRow` to align with the revised constructors.
+
+**Example**:
+
+```csharp
+[TestMethod]
+[DataRow(1, "test")]
+public void MyTestMethod(int number, string text) { ... }
+```
+
+### Timeout Settings
+
+Timeout settings are standardized across frameworks in MSTest v3. Verify and adjust any timeout configurations in existing tests for compatibility.
```

Comment:
From this description I don't get what the breaking change is. Is this point incomplete, or misplaced?